include/alpaka/math/acos/AcosUniformCudaHipBuiltIn.hpp

```diff
@@ -30,22 +30,16 @@ namespace alpaka
         {
             //! The CUDA acos trait specialization.
             template<typename TArg>
-            struct Acos<AcosUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point<TArg>::value>>
+            struct Acos<AcosUniformCudaHipBuiltIn, TArg, std::enable_if_t<std::is_floating_point_v<TArg>>>
             {
                 __device__ auto operator()(AcosUniformCudaHipBuiltIn const& acos_ctx, TArg const& arg)
                 {
                     alpaka::ignore_unused(acos_ctx);
-                    return ::acos(arg);
-                }
-            };
 
-            template<>
-            struct Acos<AcosUniformCudaHipBuiltIn, float>
-            {
-                __device__ auto operator()(AcosUniformCudaHipBuiltIn const& acos_ctx, float const& arg) -> float
-                {
-                    alpaka::ignore_unused(acos_ctx);
-                    return ::acosf(arg);
+                    if constexpr(std::is_same_v<TArg, float>)
```

Comment:
This is dangerous. What if due to changed / weird template instantiation `TArg` becomes `volatile float` - your check will be false then. I think a proper way is to check that decayed type is float, probably via a helper trait.

Reply:
See example: https://godbolt.org/z/TGfojqe41

Reply:
Btw in case our unit tests don't catch such a case, perhaps they should be extended.

Reply:
I've added a decaying check. Please re-review :-)